evap/evaluation/models.py

```diff
@@ -443,6 +444,40 @@ class State:
         verbose_name=_("wait for grade upload before publishing"), default=True
     )
 
+    @property
+    def has_exam(self):
+        return self.course.evaluations.filter(name_de="Klausur", name_en="Exam").exists()
+
+    @transaction.atomic
+    def create_exam_evaluation(
+        self,
+        exam_date: datetime,
+        evaluation_end_date: datetime,
+    ):
+        def _set_exam_evaluation_attributes(
```

Comment:
When I said that I'd vote to inline the function, I didn't mean just copying the function definition here, but actually [inline-expanding the function call](https://en.wikipedia.org/wiki/Inline_expansion) :D